lib/vault.ts

```diff
@@ -191,6 +197,164 @@ export async function listAllSubmissions(
+        FormID: {
+          S: formID,
+        },
+      };
+    }) as KeysAndAttributes["Keys"];
+
+    // DynamoDB BatchGetItem can only retrieve 100 items at a time
+    while (keys && keys.length > 0) {
+      const input = {
+        RequestItems: {
+          Vault: {
+            Keys: keys,
+          },
+        },
+        ProjectionExpression:
+          "FormID,SubmissionID,FormSubmission,ConfirmationCode,#status,SecurityAttribute,#name,CreatedAt,LastDownloadedBy,ConfirmTimestamp,DownloadedAt,RemovalDate",
+      };
+
+      const queryCommand = new BatchGetItemCommand(input);
+
+      // eslint-disable-next-line no-await-in-loop
+      const response = await documentClient.send(queryCommand);
+
+      if (response.Responses?.Vault.length) {
+        accumulatedResponses = accumulatedResponses.concat(
+          response.Responses.Vault.map((item) => {
+            return {
+              formID: item.FormID.S,
+              submissionID: item.SubmissionID.S,
+              formSubmission: item.FormSubmission.S,
+              confirmationCode: item.ConfirmationCode.S,
+              status: item.Status.S as VaultStatus,
+              securityAttribute: item.SecurityAttribute.S,
+              name: item.Name.S,
+              createdAt: item.CreatedAt.N,
+              lastDownloadedBy: item.LastDownloadedBy?.S ?? null,
+              confirmedAt: item.ConfirmTimestamp?.N ?? null,
+              downloadedAt: item.DownloadedAt?.N ?? null,
+              removedAt: item.RemovalDate?.N ?? null,
+            } as unknown as VaultSubmission;
+          })
+        );
+      }
+
+      // If there are unprocessed keys, we need to make another request
+      keys = response.UnprocessedKeys?.Vault?.Keys || [];
+    }
+
+    // Log each response retrieved
+    accumulatedResponses.forEach((item) => {
+      logEvent(
+        ability.userID,
+        {
+          type: "Response",
+          id: item.submissionID,
+        },
+        "RetrieveResponses",
+        `Retrieve selected responses for form ${formID} with ID ${item.submissionID}`
+      );
+    });
+
+    return {
+      submissions: accumulatedResponses,
+    };
+  } catch (e) {
+    logMessage.error(e);
+    return { submissions: [] };
+  }
+}
+
+/**
+ * Sets who last downloaded the Form Submission on the Vault Submission record
+ * Note that if any single update fails, the entire transaction will fail
+ * @param responses Array of responses (id, status) to update
+ * @param formID Form ID the Submission is for
+ * @param email Email address of the user downloading the Submission
+ */
+export async function updateLastDownloadedBy(
+  responses: Array<{ id: string; status: string }>,
+  formID: string,
+  email: string
+) {
+  const documentClient = connectToDynamo();
+
+  // TransactWriteItem can only update 100 items at a time
+  const asyncUpdateRequests = chunkArray(responses, 100).map((chunkedResponses) => {
+    const input: TransactWriteItemsCommandInput = {
```

Comment:
Looks like the `vault.ts` is the only place where we have not fully migrated to using `@aws-sdk/lib-dynamodb`. Maybe we should do it now that you are adding new DynamoDB requests in there. 
There are multiple examples in the code where we use it:
- In `confirm.ts` line 119 for a `BatchGetCommand` implementation
- In `confirm.ts` line 185 for a `TransactWriteCommand` implementation

Reply:
ah, did not know there were different libraries ... will update the code in this PR, but the other unrelated vault.ts queries should be updated separately (I can do that in another PR)

Reply:
Sounds good! Yeah the `lib-dynamodb` package adds a typing layer on top of the regular `client-dynamodb` which makes it easier to work with in Javascript/Typescript.

Reply:
there were some sneaky diffs in the signatures of those... took a bit of tinkering to get them working, but they've been updated